src/zyre_node.c

```diff
@@ -862,7 +879,20 @@ zyre_node_recv_peer (zyre_node_t *self)
         }
 #ifdef ZYRE_BUILD_DRAFT_API
         // TODO- CURVE- how does this trigger???
-        peer = zyre_node_require_peer (self, uuid, zre_msg_endpoint (msg), NULL);
+        char public_key[41];
+        if (self->secret_key) {
+            if (self->secret_key) {
+                char *value; // i think zhash_destroy will free this one(?)
```

Comment:
yes, lookup does not copy the value, it just returns a pointer to it

Reply:
thats what i thought- explains the odd valgrind issues i was seeing.